incubator/hnc/internal/validators/object.go

```diff
@@ -139,6 +143,30 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, op admissionv1beta
 	return o.handleInherited(op, newSource, oldSource, inst, oldInst)
 }
 
+func validateSelectorAnnot(inst *unstructured.Unstructured) string {
+	annots := inst.GetAnnotations()
+	for key, _ := range annots {
+		// for example: segs = ["propagate.hnc.x-k8s.io", "select"]
+		segs := strings.SplitN(key, "/", 2)
+		// for example: prefix = ["propagate", "hnc.x-k8s.io"]
+		prefix := strings.SplitN(segs[0], ".", 2)
```

Comment:
Will this capture anything _without_ a prefix - e.g., `hnc.x-k8s.io/foo`?

Reply:
No it won't. This function is only to catch invalid selector annotations

Reply:
Discussed offline - this is more complicated to add so let's leave it as a possible future enhancement.